risc0/build/src/docker.rs

```diff
@@ -128,7 +128,7 @@ fn create_dockerfile(
     .join(" ");
 
     let build = DockerFile::new()
-        .from_alias("build", "risczero/risc0-guest-builder:v0.17")
+        .from_alias("build", "risczero/risc0-guest-builder:v2024-01-31.1")
```

Comment:
This would still require to push to docker-hub any image with a different toolchain version.
We could add the `latest` tag to make this line constant, e.g.,  `.from_alias("build", "risczero/risc0-guest-builder:latest")` however I personally like being explicit about what version of the toolchain we are running.
A different approach would instead be to remove the installation of the toolchain from this image and install it in the next step, the one we use to build the guest. The advantage would be that we don't need to push a new image at every update, the disadvantage that the guest build would take more time.